app/controllers/flexible_criteria_controller.rb

```diff
@@ -32,15 +32,10 @@ def new
   def create
     @assignment = Assignment.find(params[:assignment_id])
     @criteria = @assignment.get_criteria
-    if @criteria.length > 0
-      new_position = @criteria.last.position + 1
-    else
-      new_position = 1
-    end
     @criterion = FlexibleCriterion.new
     @criterion.assignment = @assignment
     @criterion.max = FlexibleCriterion::DEFAULT_MAX
-    @criterion.position = new_position
+    @criterion.position = @assignment.next_criterion_position
```

Comment:
The behaviour for `Assignment#next_criterion_position` currently doesn't actually find the maximum position, which isn't good (when criteria are deleted and then new criteria are added). You should modify that method to actually compute the max position.